terra-jupyter-base/custom/google_sign_in.js

```diff
@@ -16,16 +16,56 @@ const params = {
   clusterName: ''
 };
 
-// update params with any specified in the server's config file
+// update params with any specified in the server's config file,
+// or retrieve that info from the url if we cannot access it (as is the case in terminal view)
 function updateParams() {
-  const config = Jupyter.notebook.config;
+  if (!Jupyter.notebook || !Jupyter.notebook.config || !Jupyter.notebook.config.data) {
+    console.warn('Unable to read notebook config. This is expected in terminal view, but not elsewhere. Attempting to read fallback config.');
+    readFallbackConfig();
+  } else {
+    readNotebookConfig(Jupyter.notebook.config.data);
+  }
+}
+
+function readNotebookConfig(config) {
   for (const key in params) {
-    if (config.data.hasOwnProperty(key)) {
-      params[key] = config.data[key];
+    if (config.hasOwnProperty(key)) {
+      params[key] = config[key];
     }
   }
 }
 
+// here we attempt to parse the url for the googleProject and the clusterName
+function readFallbackConfig() {
+  const url = window.location.href;
+  const initialSearch = 'proxy/';
+
+  const projectSubstringStartLocation = url.search(initialSearch) + initialSearch.length;
+  const projectSubstring = url.substring(projectSubstringStartLocation, url.length);
+  const projectEndLocation = projectSubstring.search('/');
+  const googleProject = projectSubstring.substring(0, projectEndLocation);
+
+  // we add 1 for the slash between project and cluster
+  const clusterSubstring = projectSubstring.substring(projectEndLocation + 1, projectSubstring.length);
+  const clusterEndLocation = clusterSubstring.search('/');
+  const clusterName = clusterSubstring.substring(0, clusterEndLocation);
+
+  console.info(`Attempted to parse the url for a fallback configuration. 
+                         Found googleProject: '${googleProject}' and clusterName '${clusterName}'`);
+
+  params.googleProject = googleProject;
+  params.clusterName = clusterName;
+
+  fallbackReadNotebookConfig(googleProject, clusterName);
+}
+
+function fallbackReadNotebookConfig(googleProject, clusterName) {
+  const url = `/notebooks/${googleProject}/${clusterName}/api/config/notebook`;
```

Comment:
Nitpick: can we use this URL format?
```
/proxy/googleProject/clusterName/jupyter/api/config/notebook
```
The /notebooks one is technically deprecated, I don't think any clients are using it anymore.